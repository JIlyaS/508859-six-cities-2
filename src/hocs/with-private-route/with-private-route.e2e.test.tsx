import * as React from 'react';
import * as Enzyme from "enzyme";
import * as Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import withPrivateRoute from './with-private-route';

Enzyme.configure({adapter: new Adapter()});

const MockComponent = () => <div />;
const MockComponentWrapped = withPrivateRoute(MockComponent, true, `/login`);

describe(`withPrivateRoute HOC work correct`, () => {
  it(`PrivateRoute component is correct`, () => {
    const wrapper = Enzyme.shallow(<MockComponentWrapped />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
