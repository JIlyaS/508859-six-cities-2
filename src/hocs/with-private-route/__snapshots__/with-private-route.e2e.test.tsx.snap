// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPrivateRoute HOC work correct PrivateRoute component is correct 1`] = `
<Route
  render={[Function]}
/>
`;
